Classes/Controller/View3DController.php

```diff
@@ -41,21 +41,23 @@ public function mainAction()
             return '';
         } else {
             $url = $this->document->getDoc()->getFileLocation($this->document->getDoc()->physicalStructureInfo[$this->document->getDoc()->physicalStructure[1]]['files']['DEFAULT']);
+            $proxy = '';
             if ($this->settings['useInternalProxy']) {
                 // Configure @action URL for form.
                 $uri = $this->uriBuilder->reset()
                     ->setTargetPageUid($GLOBALS['TSFE']->id)
                     ->setCreateAbsoluteUri(!empty($this->settings['forceAbsoluteUrl']) ? true : false)
                     ->setArguments([
                         'eID' => 'tx_dlf_pageview_proxy',
-                        'url' => urlencode($url),
-                        'uHash' => GeneralUtility::hmac($url, 'PageViewProxy')
+                        'url' => $url,
+                        'uHash' => 'View3DController'
```

Comment:
Why did you remove the uHash? This makes Kitodo.Presentation vulnerable to XSS attacks. Please keep this the same as in the PageView plugin. (You can even keep the same message `PageViewProxy`.)
```suggestion
                        'uHash' => GeneralUtility::hmac($url, 'PageViewProxy')
```

Reply:
Because if the file name of model is changed than proxy is failing. You know this problem with the incorrect names of the model and then changed suffix. After suffix is adjusted our URL needs to also change and then it all doesn't work anymore.

Reply:
OK, then @thedworak needs to find a fix for this issue even more urgently. We can't open up a security issue in Kitodo.Presentation just because Drupal isn't able to give the right name of an uploaded file. @thedworak has to make sure that the METS file contains a valid file URL - either by finding a way to get the right file name from Drupal in the first place or by implementing some kind of transparent redirecting on the Drupal side when calling a file without its suffix. We have to be able to rely on the path given in the METS file!

Reply:
Maybe from the beginning: Why you don't have a right filename for downloading?

Reply:
The file name can be modified inside the JS if XML contains the incorrect one, right? Drupal problem with the incorrect model file naming. If suffix is changing then proxy link is immediately invalidated.